app/config.py

```diff
@@ -180,8 +187,11 @@ def __getattr__(self, name):
     def __getitem__(self, name):
         """Gets a configuration setting for this repository.  Looks for a
         repository-specific setting, then falls back to a global setting."""
-        return get_for_repo(self.repo, name)
+        if name in self.entries:
+            return self.entries[name]
+        elif self.global_config:
+            return self.global_config[name]
```

Comment:
It won't raise an exception. Since this function overrides accesses like `config[name]`, line 193 is basically a recursive call into this same function for the global config object. The global config won't have a pointer to any other config (i.e., `self.global_config` will evaluate as false) and it'll jump down to 'return None`.
I added a test for this to be safe (`test_no_exception_for_unset_key` in `server_test_cases/config_tests.py`).

Reply:
Oh I see, it's a Config instance not a dict.